core/src/wsv.rs

```diff
@@ -625,58 +652,103 @@ impl WorldStateView {
     /// Get `Account` and pass it to closure to modify it
     ///
     /// # Errors
-    /// Fails if there is no domain or account
+    /// Forward errors from [`Self::modify_account_multiple_events`]
     pub fn modify_account(
         &self,
         id: &AccountId,
         f: impl FnOnce(&mut Account) -> Result<AccountEvent, Error>,
     ) -> Result<(), Error> {
-        self.modify_domain(&id.domain_id, |domain| {
+        self.modify_account_multiple_events(id, move |account| f(account).map(std::iter::once))
+    }
+
+    /// Get `Account` and pass it to closure to modify it
+    ///
+    /// # Errors
+    /// - If there is no domain or account
+    /// - Forward errors from `f`
+    pub fn modify_account_multiple_events<I: IntoIterator<Item = AccountEvent>>(
+        &self,
+        id: &AccountId,
+        f: impl FnOnce(&mut Account) -> Result<I, Error>,
+    ) -> Result<(), Error> {
+        self.modify_domain_multiple_events(&id.domain_id, |domain| {
             let account = domain
                 .account_mut(id)
                 .ok_or_else(|| FindError::Account(id.clone()))?;
-            f(account).map(DomainEvent::Account)
+            f(account).map(|events| events.into_iter().map(DomainEvent::Account))
         })
     }
 
     /// Get `Asset` by its id
     ///
     /// # Errors
-    /// Fails if there are no such asset or account
+    /// Forward errors from [`Self::modify_asset_multiple_events`]
     #[allow(clippy::missing_panics_doc)]
     pub fn modify_asset(
         &self,
         id: &<Asset as Identifiable>::Id,
         f: impl FnOnce(&mut Asset) -> Result<AssetEvent, Error>,
     ) -> Result<(), Error> {
-        self.modify_account(&id.account_id, |account| {
+        self.modify_asset_multiple_events(id, move |asset| f(asset).map(std::iter::once))
+    }
+
+    /// Get `Asset` by its id
```

Comment:
function names start with `modify` but function descriptions are a bit different:

1) `and modify it with f`
2) `with an ability to modify it`
3) `and pass it to closure to modify it`
4) nothing

I'm trying to understand whether this is intentional: do all four cases mean different things? if not, then the descriptions should not be different

Reply:
Use case of all this functions is the same.
We are passing object (world/asset/account/...) into closure `f` and receive bunch of events or error from it. 
We cant know what `f` is actually do with provided object (`f` can modify it or leave it as is).